shared/utils/utils.h

```diff
@@ -23,6 +23,14 @@ template <typename Container> struct is_vector : std::false_type { };
 template <typename... Ts> struct is_vector<std::vector<Ts...> > : std::true_type { };
 // TODO: figure out how to write an is_vector_v that compiles properly?
 
+#define MACRO_WRAP(x) do { \
+    x; \
+} while(0)
+
+#define CRASH_IF_NOT(arg) MACRO_WRAP( \
+    if (!(arg)) SAFE_ABORT(); \
```

Comment:
Shouldn't actually need to wrap a one line'd if statement, since it'll still functionally work when used _as_ a one liner (the if will just nest to the next line as intended anyways).
Typically, wraps are for multi-line defines.

Reply:
In addition, this new method should be used as a replacement in _many_ locations, or a similar macro should be made for simply returning false with a critical log. That would simplify a lot of the redundancy/boilerplate-y-ness of customui

Reply:
`if (blah) if (blah) blah;` compiles fine? ok

Reply:
nah fam sounds dangerous https://stackoverflow.com/questions/11734604/if-block-without-curly-braces-makes-subsequent-else-if-nested

Reply:
> In addition, this new method should be used as a replacement in _many_ locations, or a similar macro should be made for simply returning false with a critical log. That would simplify a lot of the redundancy/boilerplate-y-ness of customui

done via RET(_0)_UNLESS